homestar-runtime/tests/network/notification.rs

```diff
@@ -184,3 +184,382 @@ fn test_connection_notifications_integration() -> Result<()> {
+            } else {
+                panic!("Node one did not establish a connection with node two in time.")
+            }
+        }
+
+        kill_homestar(proc_guard2.take(), None);
+
+        // Poll for connection closed message
+        loop {
+            if let Ok(msg) = sub1.next().with_timeout(Duration::from_secs(30)).await {
+                let json: serde_json::Value =
+                    serde_json::from_slice(&msg.unwrap().unwrap()).unwrap();
+
+                if json["type"].as_str().unwrap() == "network:connectionClosed" {
+                    break;
+                }
+            } else {
+                panic!("Connection between node one and node two did not close in time.")
+            }
+        }
+
+        // Poll for outgoing connection error message
+        loop {
+            if let Ok(msg) = sub1.next().with_timeout(Duration::from_secs(30)).await {
+                let json: serde_json::Value =
+                    serde_json::from_slice(&msg.unwrap().unwrap()).unwrap();
+
+                if json["type"].as_str().unwrap() == "network:outgoingConnectionError" {
+                    break;
+                }
+            } else {
+                panic!("Connection between node one and node two did not close in time.")
+            }
+        }
+
+        // Poll for outgoing connection error message
+        loop {
+            if let Ok(msg) = sub1.next().with_timeout(Duration::from_secs(30)).await {
+                let json: serde_json::Value =
+                    serde_json::from_slice(&msg.unwrap().unwrap()).unwrap();
+
+                if json["type"].as_str().unwrap() == "network:outgoingConnectionError" {
+                    break;
+                }
+            } else {
+                panic!("Connection between node one and node two did not close in time.")
+            }
+        }
+
+        let homestar_proc2 = Command::new(BIN.as_os_str())
+            .env(
+                "RUST_LOG",
+                "homestar=debug,homestar_runtime=debug,libp2p=debug,libp2p_gossipsub::behaviour=debug",
+            )
+            .arg("start")
+            .arg("-c")
+            .arg(config2.filename())
+            .arg("--db")
+            .arg(&proc_info2.db_path)
+            .stdout(Stdio::piped())
+            .spawn()
+            .unwrap();
+        let proc_guard2 = ChildGuard::new(homestar_proc2);
+
+        // Poll for connection established message
+        loop {
+            if let Ok(msg) = sub1.next().with_timeout(Duration::from_secs(30)).await {
+                let json: serde_json::Value =
+                    serde_json::from_slice(&msg.unwrap().unwrap()).unwrap();
+
+                if json["type"].as_str().unwrap() == "network:connectionEstablished" {
+                    break;
+                }
+            } else {
+                panic!("Node one did not redial node two in time.")
+            }
+        }
+
+        kill_homestar(proc_guard1.take(), None);
```

Comment:
@QuinnWilton I don't think we need the kills at all b/c they will be killed automatically on drop, due to the guard.

Reply:
Ah whoops, I forgot there was a second test to remove these from! Fixing it now.